Editor/CoreAPI/StateManager.cs

```diff
@@ -0,0 +1,41 @@
+﻿using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+
+namespace Editor.CoreAPI
+{
+    public class StateManager
+    {
+        public CoreApi CoreApi { get; }
+
+        public GameLiftFleetManager FleetManager { get; set; }
+        public GameLiftComputeManager ComputeManager { get; set; }
+
+        public IAmazonGameLiftClientWrapper GameLiftWrapper { get; private set; }
+
+        public IAmazonGameLiftClientFactory AmazonGameLiftClientFactory { get; }
+
+        private string _selectedProfile;
+
+        public string SelectedProfile
+        {
+            get => _selectedProfile;
+            private set => SetupWrapper(value);
+        }
+
+        public StateManager(CoreApi coreApi)
+        {
+            CoreApi = coreApi;
+            AmazonGameLiftClientFactory = new AmazonGameLiftClientFactory(coreApi);
+
+            SelectedProfile = coreApi.GetSetting(SettingsKeys.CurrentProfileName).Value;
+        }
+
+        private void SetupWrapper(string profileName)
+        {
+            _selectedProfile = profileName;
```

Comment:
We probably need a null check on this. If the profile name is set to null, we don't want wrapper/managers

Reply:
I'm assuming we won't have a need to "unset" the profile name. Like once a customer has selected a profile, there's no use case to remove that setting. If we did we'd want this null check to set all these to null, but I believe we're good so moving on